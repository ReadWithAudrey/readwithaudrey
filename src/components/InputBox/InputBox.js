import React from 'react'
import { StyledInputBox } from './InputBox.style'
import PropTypes from 'prop-types'

const InputBox = props => {
  if (props.required) {
    return (
      <StyledInputBox
        placeholder={props.placeholder}
        onChange={props.onChange}
        name={props.name}
        type={props.type}
        required
      />
    )
  } else {
    return (
      <StyledInputBox
        placeholder={props.placeholder}
        onChange={props.onChange}
        name={props.name}
        type={props.type}
      />
    )
  }
}

InputBox.propTypes = {
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
  name: PropTypes.string,
  type: PropTypes.string,
}

export default InputBox
