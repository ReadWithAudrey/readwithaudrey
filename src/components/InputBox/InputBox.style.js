import styled from 'styled-components'

export const StyledInputBox = styled.input.attrs({
  className: 'montserrat mv2 pa2 w-100',
})`
  background: #fff;
  display: block;
  border: #d3d3d3 1px solid;
  border-radius: 10px;
  box-sizing: border-box;
`
