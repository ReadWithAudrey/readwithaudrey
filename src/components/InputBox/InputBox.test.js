import React from 'react'
import renderer from 'react-test-renderer'
import InputBox from './InputBox'

describe('InputBox component snapshot test', () =>
  it('renders correctly', () => {
    const tree = renderer.create(<InputBox />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
