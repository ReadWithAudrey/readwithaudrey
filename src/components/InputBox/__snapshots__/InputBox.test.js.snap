// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox component snapshot test renders correctly 1`] = `
<input
  className="sc-bdVaJa montserrat mv2 pa2 w-100 dtdCXZ"
/>
`;
